client/src/services/informationManager.ts

```diff
@@ -0,0 +1,155 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+import { MsAlertModal, MsAlertModalConfig, MsReportTheme } from '@/components/core';
+import { ToastManager } from '@/services/toastManager';
+import { modalController } from '@ionic/vue';
+import { v4 as uuid4 } from 'uuid';
+import { NotificationManager } from '@/services/notificationManager';
+
+export const InformationKey = 'information';
+
+export enum InformationLevel {
+  Info = 'INFO',
+  Success = 'SUCCESS',
+  Warning = 'WARNING',
+  Error = 'ERROR',
+}
+
+export interface IInformation {
+  message: string;
+  level: InformationLevel;
+  data?: object
+}
+
+export class Information {
+  id: string;
+  message: string;
+  level: InformationLevel;
+  data?: object;
+  constructor({ message, level, data = {} }: IInformation) {
+    this.id = uuid4();
+    this.message = message;
+    this.level = level;
+    this.data = data;
+  }
+
+  get theme(): MsReportTheme {
+    switch (this.level) {
+      case InformationLevel.Info:
+        return MsReportTheme.Info;
+      case InformationLevel.Success:
+        return MsReportTheme.Success;
+      case InformationLevel.Warning:
+        return MsReportTheme.Warning;
+      case InformationLevel.Error:
+        return MsReportTheme.Error;
+    }
+  }
+}
+
+export interface InformationOptions {
+  trace?: boolean;
+}
+
+export enum InformationPresentationMode {
+  Toast,
+  Modal,
+  Notification,
+  ToastAndNotification,
+  ModalAndNotification,
+}
+
+export interface InformationToastOptions extends InformationOptions {
```

Comment:
Not sure about this interface. It is used in `showToast()`, but in `present()` you display a toast using `InformationOptions` only.

Reply:
normally since it extends the `InformationOptions` it should work anyway, or maybe I need to update to:
`    options: InformationOptions | InformationToastOptions,`

Reply:
It shouldn't in a normal object language because `InformationOptions` does not have a `duration` attribute. Updating it is probably better for now, we'll check later if we find something better.